### PR TITLE
calico: Update calico to v1.25.2 for each of the providers

### PR DESCRIPTION
**Pull request by brianmcarey**
**What this PR does / why we need it**:

Calico v1.25 or later is required to help enable s390x kubevirtci providers

There were issues seen when updating to v1.27.2[1] 

I provisioned this locally and ran the network tests in question against it which passed. 

[1] https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/13041/pull-kubevirt-e2e-k8s-1.30-sig-network/1845730975342923776

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/cc @qinqon @ormergi @xpivarc 

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

**Review by xpivarc — COMMENTED** [review]
/approve
/hold for network team

File: cluster-provision/k8s/1.29/extra-pre-pull-images [hunk 1]
```diff
@@ -2,10 +2,10 @@ quay.io/kubevirtci/install-cni:1.15.0
...
 quay.io/kubevirtci/pilot:1.15.0
 quay.io/kubevirtci/proxyv2:1.15.0
-quay.io/calico/cni:v3.18.0
-quay.io/calico/kube-controllers:v3.18.0
-quay.io/calico/node:v3.18.0
-quay.io/calico/pod2daemon-flexvol:v3.18.0
+quay.io/calico/cni:v3.25.2
+quay.io/calico/kube-controllers:v3.25.2
+quay.io/calico/node:v3.25.2
+quay.io/calico/pod2daemon-flexvol:v3.25.2
```
Comment (chandramerla) [inline cluster-provision/k8s/1.29/extra-pre-pull-images:8]: @brianmcarey Do we need to keep "quay.io/calico/pod2daemon-flexvol:v3.25.2" entry in cluster-provision/k8s/1.*/extra-pre-pull-images files? As I see it is no more being referred in the newer cni.do-not-change.yaml files, unlike with v3.18.0 cni.do-not-change.yaml was having a reference like below:

>          - name: flexvol-driver
>            image: docker.io/calico/pod2daemon-flexvol:v3.18.0

Not sure if this image is required else where than cni yaml file. In earlier PR to bump calico to 3.27.2 #1281, I removed the line "quay.io/calico/pod2daemon-flexvol:v3.25.2" from extra-pre-pull-images.

Comment (brianmcarey) [reply cluster-provision/k8s/1.29/extra-pre-pull-images]: Good catch - thanks @chandramerla  - removed.

**Pull request closed without merging**